blog/content/post/2021-07-13-ftpm-vs-dtpm.md

```diff
@@ -0,0 +1,125 @@
+registers. Without going into too much detail, it can verify the integrity of
+every component in the boot process and allows the establishment of a trusted
+computing platform.
+
+While TPMs began as dedicated security chips, as they picked up adoption,
+silicon vendors started implementing them in chipsets and eventually in firmware.
+A firmware based implementation is commonly called a Firmware TPM, or fTPM for
+short, runs in the CPU's trusted execution environment, doesn't take up any
+extra space on the board which is great for space-constrained boards, and is
+drastically cheaper to implement than a dedicated chip. A hardware based
+implementation in a discrete package is now called Discrete TPM, or dTPM.
+
+You may have also heard about sTPM, a software implementation of TPM running in
+userspace, but as it's not at all isolated from the host OS (without the use of
+technologies like Intel SGX, at least) it doesn't offer any real security
+benefits. It's mostly just useful for development purposes.
+
+## HW vs FW
+
+One might be inclined to say that dTPM is more secure because it's a dedicated
+device, but it's not that simple and what will work best for you depends on a
+varuiery of factors. A discrete TPM is designed to be tamper resistant,
+provides a higher degree of isolation, and can be certified by TCG (in fact
+most of them are). It is, however, slower than other implementations
+(though that is by design) and is commonly connected to the LPC bus which can be
+sniffed. Side channel attacks are also potentially easier to perform when
+the TPM is located on a separate chip.
+
+fTPMs present a different set of security considerations: for example if the
+firmware is writable for an attacker it could be replaced with a vulnerable
+version. To prevent this, countermeasures such as firmware signing and fTPM
+clear on rollback need to be implemented. fTPMs also generally aren't certified
+as vendors don't generally let certifying agencies look at their sources.
+Lastly, fTPMs may share the attack surface with management coprocessors like
+Intel ME and AMD PSP.
+
+On the other hand, fTPMs are much less vulnerable to sniffing by being contained
+within the CPU package. There are no exposed connections between the TPM and CPU
+like there is with a dTPM, which is connected to the LPC bus, so sniffing keys
+would require debugging the CPU with JTAG or even physically decapping it to
+access internal traces.
+
+We also have to consider the particular functions provided by TPM. For example,
+how is secure storage implemented? dTPMs have tamper resistant NVRAM, while
+with fTPMs there are a couple of possibilities:
+
+- TrustZone relies on trusted storage provisions in eMMC controllers
+- Intel PTT can store secrets in an encrypted portion of the SPI
+flash
+
+Then we have to consider the memory security facilities:
+
+- TZ defines a separate region for the "normal world" in which the usual
```

Comment:
```suggestion
- TrustZone defines a separate region for the "normal world" in which the usual
```
Let's avoid abbreviations especially those not obvious or not introduced earlier in blog post.